### PR TITLE
Add unit-tests for parenchecker

### PR DESCRIPTION
**Pull request by dorlet-cloud**

### CODE REVIEW

File: src/stack/examples/parenchecker.test.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    const result = validParens('')
+    expect(result).toBe(true)
+  }),
+  
+  it('should return true if open & close parentheses match', () => {
+    const result = validParens('({(<>)})')
+    expect(result).toBe(true)
+  }),
+
+  it('should return false if open & close parantheses do not match', () => {
```
Comment (misrule) [inline src/stack/examples/parenchecker.test.ts:15]: It's fine, but in a real-world codebase I would do more, varied checks for the failure case to ensure there are no corner-cases that trick the function. What if the number of each bracket type *does* match, but the ordering is wrong? Does space matter? Things like that.  

Important to remember that unit tests should document the expected behavior of your code. If spaces shouldn't matter, then describe that behavior. You might know from the code that spaces get stripped out or whatever - but what if the implementation changes in the future? How will they know how the function is supposed to behave? Whoever makes the changes might not strip spaces - breaking the code (clients will break), but all the unit tests pass because correct behavior was not fully described.

**Review by misrule — APPROVED** [review]
